design/authenticators-status/authn_status_general.md

```diff
@@ -0,0 +1,113 @@
+# Feature Overview
+
+***As a*** Conjur operator\
+***I'd like to*** know that an authenticator is configured correctly\
+***So that*** I can complete the configuration properly if it's incomplete (or remove
+it if it's not needed).
+
+This feature lets the person who configures an authenticator to get an immediate feedback
+on the configuration, before any user needs to run an authentication request.
+
+# Implementation Details
+
+![alt text](authn-status-flow.jpeg "Authenticator Status API flow")
+
+We will create a new route in `routes.rb` for `/authenticators/:authenticator(/:service_id)/:account/status`.
+This route will lead to `AuthenticateController` which will consist of a new `status` method.
+In this method we will call a new CommandClass `Authentication::Status`. This class's
+`call` method will perform the general checks that apply to all authenticators
+and those specific to each authenticator:
+
+- The requesting user has access to the authenticator status route webservice
+    - more info in the Security section of the [epic](https://github.com/cyberark/conjur/issues/1062)
+- The authenticator is implemented
+- The account exists
+    - although a wrong account in an `/authenticate` request will indicate a client error and
+    not a server error, we still need to verify it exists in order to verify that the webservice exists
+    under that account
+- The webservice exists
+- The authenticator is enabled in the ENV
+- [Specific authenticator requirements](#implementing_specific_authenticator_status_check)
+ 
+***Note:*** The default authenticator (`authn`) is always healthy
```

Comment:
@orenbm what if it's not enabled in the authenticators environment variable?
For example, in K8s, I would not enable `authn` for the followers

Reply:
@rafis3 this is something that i noticed sometime earlier. even if you don't mention `authn` in the CONJUR_AUTHENTICATORS env var it is still enabled. we should double-check this because it contradicts the documentation

Reply:
so if we want to have the ability to disable `authn` then we have a bug

Reply:
Yes, I think I noticed this behavior as well. In that case, yes it is a bug.

Reply:
@rafis3 thanks, i opened a bug - https://github.com/cyberark/conjur/issues/1076